Descent3/CMakeLists.txt

```diff
@@ -292,18 +292,25 @@ endif()
 
 file(GLOB_RECURSE INCS "../lib/*.h")
 
-add_executable(Descent3
-  WIN32
-  ${HEADERS} ${CPPS} ${PLATFORM_CPPS} ${INCS}
-)
+if(ANDROID)
+  add_library(Descent3
```

Comment:
Oh, interesting. You're supposed to build it as a lib for Android?

Reply:
yep. The Android application lifecycle is pretty different from what we normally think of. I won't dive into the details unless you're interested, but the bottom line is that any application-provided native code must be `dlopen`-ed (effectively) into an existing Java process. That Java process can then call into functions exported from the library and we can begin our work from there.